app/models/keyword.rb

```diff
@@ -0,0 +1,5 @@
+# frozen_string_literal: true
+
+class Keyword < ApplicationRecord
+  belongs_to :user, inverse_of: :keywords
```

Comment:
According to our compass conventions, should we use soft delete here. 
<img width="943" alt="Screen Shot 2564-06-25 at 15 45 16" src="https://user-images.githubusercontent.com/11224179/123397216-5de0f280-d5cc-11eb-9164-1d1128038bc4.png">

```
   Always soft-delete relational data for integrity and audit purposes. Gems such as paranoia and discard provide ready-to-use solutions.
```

Reply:
Yes, makes perfect sense. 

As there will be no delete/destroy action exposed for keywords, I will try to handle it on the only scenario when a user "cancel his account" (so that it soft-delete the user, but it can keep keywords as is).

Would that be ok @ankitkalia1195 ? :) Thanks!

Reply:
ok, soft delete for Users implemented and tested. 
Do you think we need to add tests @ankitkalia1195 ? 
Candidates for tests:
- Once deleted, a user get logged out
- Once deleted, a user cannot log in anymore
- Once deleted, a user cannot reset his password

it's build-in with devise and discard, so I'm not sure if tests are needed in here~
Thanks for your advises :)

Reply:
@malparty It seems you have again overcomplicated things, but its okay. For testing soft delete in models, you could have just tested them via rails console. We add soft delete while designing the model class, does not necessary mean that we should have some UI to test it.

Reply:
The main motive is to just ensure that data does not get deleted from database, whether is user or keyword. So usually when working on projects, we add soft delete to models straight away.

Reply:
And can we keep using paranoia instead of discard, as it is in the rails template. With paranoia, you dont need to override anything like `devise` as well. Just calling `user.delete` will trigger soft delete

Reply:
For reference, you can check project mia. https://github.com/nimblehq/mia-web/tree/development/app/models. Here we have `acts_as_paranoid` in all models

Reply:
yes, the idea was to practice, there is a high chance that in a future project I get the need for a user soft-delete feature  ^^, 
For the choice of `discard` over `paranoia`, it's due to the notice heading `paranoia` readme: 
> paranoia has some surprising behaviour (like overriding ActiveRecord's delete and destroy) and is not recommended for new projects. See discard's README for more details.

I'll remember to add a soft-delete even when not needed -- with the intend to prevent unexpected hard-delete into future features.

Reply:
@malparty Sure, we can use `discard` as well, altough in most of our projects we still use `paranoia`.

Reply:
Basically `paranoia` is easier to use and does everything `magically`, `discard` is more explicit, and you have to write more code in order to achieve the result. In our rails template we still prefer `paranoia`, but you can bring it to a vote if it should be `discard`

Reply:
Ok, yes I saw `paranoia` replace the destroy/delete methods so it has this "magic rails" feel :) 
As I started with `discard` and as you said, I already complicated things, then I've just added the soft delete to keywords -- as per convention.

| Some little rails console test: | It fills the discarded_at date: | 
|---|---|
| ![image](https://user-images.githubusercontent.com/77609814/123746406-cdf8bc80-d8db-11eb-8d23-e5fc6cc5d983.png) | ![image](https://user-images.githubusercontent.com/77609814/123746492-e9fc5e00-d8db-11eb-93e6-3094919283df.png) |

And as the scope of keywords is `kept`, it does not display the keywords anymore:
![image](https://user-images.githubusercontent.com/77609814/123746554-00a2b500-d8dc-11eb-9761-cfda3b449f07.png)